v2/options.go

```diff
@@ -0,0 +1,48 @@
+package car
+
+type ReadOptions struct {
+	V1NullPadding bool
+
+	BlockstoreUseWholeCIDs bool
+}
+
+type ReadOption func(*ReadOptions)
+
+func (ReadOption) readWriteOption() {}
+
+var _ ReadWriteOption = ReadOption(nil)
+
+type WriteOptions struct {
+	CarV1Padding uint64
+	IndexPadding uint64
+
+	BlockstoreAllowDuplicatePuts bool
+}
+
+type WriteOption func(*WriteOptions)
+
+func (WriteOption) readWriteOption() {}
+
+var _ ReadWriteOption = WriteOption(nil)
+
+type ReadWriteOption interface {
+	readWriteOption()
+}
+
+func AllowV1NullPadding(o *ReadOptions) {
```

Comment:
`TreatZeroLengthFrameAsEOF` or something like that is more descriptive of what the code actually does if this option is set. The fact that it's used as a way to add "padding" is I think external to CAR v1.

Reply:
Yup agreed.